CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+Installation
+============
+
+```shell
+$ git clone https://github.com/SUSE-Enceladus/suse-cloud-image-name-parser.git
+$ cd suse-cloud-image-name-parser
+
+# Activate virtual Environment then install
+# and dev dependences in editable mode
+$ pip install -e .[dev]
+```
+
+Dev Requirements
+================
+
+- bumpversion
+- pip\>=7.0.0
+
+Testing Requirements
+====================
+
+- coverage
+- flake8
+- pylint 
+- pytest-cov
+
+Contribution Checklist
+======================
+
+- All patches must be signed. [Signing Commits](#signing-commits)
+- All contributed code must conform to flake8. [Code Style](#code-style)
+- All new code contributions must be accompanied by a test.
+    - Tests must pass and coverage remain above 90%. [Unit & Integration Tests](#unit-&-integration-tests)
+- Follow Semantic Versioning. [Versions & Releases](#versions-&-releases)
+
+Versions & Releases
+===================
+
+**suse-cloud-image-name-parser** adheres to Semantic versioning; see <http://semver.org/> for
+details.
+
+[bumpversion](https://pypi.python.org/pypi/bumpversion/) is used for
+release version management, and is configured in `setup.cfg`:
+
+```shell
+$ bumpversion major|minor|patch
+$ git push
+$ git push --tags
+```
+
+Bumpversion bumps the version and also creates a new tag for the repo.
+
+Unit & Integration Tests
+========================
+
+All tests should pass and test coverage should remain above 90%.
+
+The tests and coverage can be run directly via pytest.
+
+```shell
+$ pytest --cov=imagenameparser
+```
+
+Testing with tox
+================
+
+Requirements
+------------
+
+- tox
+- tox-pyenv (testing multiple Python versions 3.4+)
+
+Setup
+-----
+
+- Install tox
+    
+    ```shell
+    $ pip install .\[tox\]
+    ```
+
+- Setup pyenv
+
+    See the [pyenv doc](https://github.com/pyenv/pyenv#installation) for
+    more info.
+
+- Run tox
+
+    ```shell
+    $ tox (test all versions 3.4+)
+
+    # or
+    $ tox -e py{version}
+    $ tox -e py36
+    ```
+
+Code Style
+==========
+
+Source should pass flake8 and pep8 standards.
+
+```shell
+$ flake8 imagenameparser 
```

Comment:
Could offer an alternative here to run the `tox -e pep8` command since that should be the same version as the CI test runs, in case someone is running a system flake8 package rather than the version in the activated developer mode venv.